### PR TITLE
add timeline pane and message handler

### PR DESCRIPTION
**Pull request by am-zhang**
# Description

Added a new empty Timeline pane to the top right of the Dashboard UI. Sends and receives a handshake message from the agent using a simple handler. 

Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

Before: No Timeline tab on the top right of the Dashboard.
After:
Screen Shot 2021-05-25 at 3 17 34 PM

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

Screen Shot 2021-05-24 at 3 02 18 PM
Screen Shot 2021-05-24 at 3 03 10 PM

# Testing

Please describe the tests that you ran to verify your changes and how we can reproduce.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

File: base_agent/loco_mc_agent.py [hunk 1]
```diff
@@ -138,6 +138,11 @@ def send_text_command_to_agent(sid, command):
             }
             sio.emit("setChatResponse", payload)
 
+        @sio.on("receiveHandshake")
+        def receive_handshake(sid, handshake):
```
Comment (snyxan) [inline base_agent/loco_mc_agent.py:142]: The `handshake` argument is never used in this function. It's directly assigned a value and sent back to dashboard. 
Generally if one argument is never used, then do not pass it. In this case, you can verify the handshake message first (e.g. check if it matches a given pattern like 'ping'), log it, then reply back with another message (e.g. 'pong' or received).

File: dldashboard/web/src/StateManager.js [hunk 2]
```diff
@@ -53,6 +54,7 @@ class StateManager {
       { msg: "", failed: false },
       { msg: "", failed: false },
     ],
+    handshake: "",
```
Comment (snyxan) [inline dldashboard/web/src/StateManager.js:57]: StateManager maintains all states for various components. If you need to put a field here, make sure to prefix it with component name, i.e. timeline.

File: dldashboard/web/src/StateManager.js [hunk 3]
```diff
@@ -66,6 +68,7 @@ class StateManager {
     this.processRGB = this.processRGB.bind(this);
     this.processDepth = this.processDepth.bind(this);
     this.processObjects = this.processObjects.bind(this);
+    this.returnHandshake = this.returnHandshake.bind(this);
```
Comment (snyxan) [inline dldashboard/web/src/StateManager.js:71]: same as above

File: dldashboard/web/src/StateManager.js [hunk 4]
```diff
@@ -172,6 +176,15 @@ class StateManager {
     });
   }
 
+  returnHandshake(res) {
```
Comment (snyxan) [inline dldashboard/web/src/StateManager.js:179]: same as above.

**Review by snyxan — COMMENTED** [review]
Thanks for creating this PR @am-zhang ! 

Overall it looks great, you implemented all the requested features and checked in the build files, great job!

You've attached screenshots which is important for UI changes, keep doing that.

Some tips:
- Keep your PR small and easy to review. Try not to involve too many changed files or lines. For this PR it's okay because you are checking in lots of build files. Next time you can push functional changes, request for reviews, then check in build files.
- Mention which commit you want the reviewer to review if only part of them needs to be reviewed.
- Ask for more reviewers who have context on the changes. It will help your PR get approved faster. If you are not sure who to request, you can blame the files or check the change history. For example, for this PR you can also add Soumith or Kavya as reviewer.

**Comment by snyxan** [comment]
Also, can you put `timeline.js` to a separate folder like Memory/TeachApp? As you develop this component it will grow larger and more component-specific files will be checked in.

**Comment by am-zhang** [comment]
Thanks for the great feedback! I've implemented these changes in [d1d82c3](https://github.com/facebookresearch/droidlet/pull/437/commits/d1d82c3d194be7ecdda5e1d61aa78984a8629a79). Let me know how this looks!

**Review by snyxan — APPROVED** [review]
LGTM, now ship it!

Tip: use 'squash and merge' option when you merge PRs into main branch with a descriptive message about what this PR does. Only use regular merge (should be less frequently) if you think keeping every single commit of that PR in history is important.

**Pull request merged**